fe/fe-core/src/main/java/com/starrocks/catalog/Dictionary.java

```diff
@@ -0,0 +1,321 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.common.util.TimeUtils;
+import com.starrocks.persist.gson.GsonPostProcessable;
+import com.starrocks.persist.gson.GsonUtils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Dictionary implements Writable, GsonPostProcessable {
+    // Dictionary properties
+    public static final String PROPERTIES_DICTIONARY_WARMING_UP = "dictionary_warming_up";
+    public static final String PROPERTIES_DICTIONARY_MEMORY_LIMIT = "dictionary_memory_limit";
+    public static final String PROPERTIES_DICTIONARY_REFRESH_TIME = "dictionary_refresh_time";
+
+    public enum DictionaryState {
+        UNINITIALIZED,
+        REFRESHING,
+        FINISHED,
+        CANCELLED
+    }
+
+    @SerializedName(value = "dictionaryId")
+    private long dictionaryId;
+    @SerializedName(value = "dictionaryName")
+    private String dictionaryName;
+    @SerializedName(value = "dbName")
+    private String dbName;
+    @SerializedName(value = "queryableObject")
+    private String queryableObject;
+
+    @SerializedName(value = "dictionaryKeys")
+    private List<String> dictionaryKeys = Lists.newArrayList();
+    @SerializedName(value = "dictionaryValues")
+    private List<String> dictionaryValues = Lists.newArrayList();
+    @SerializedName(value = "nextSchedulableTime")
+    private AtomicLong nextSchedulableTime = new AtomicLong(Long.MAX_VALUE); // ms
+
+    private Map<String, String> properties = Maps.newHashMap();
+
+    private long lastSuccessRefreshTime = 0;
+    private long lastSuccessFinishedTime = 0;
+    private DictionaryState state = DictionaryState.UNINITIALIZED;
+    private DictionaryState stateBeforeRefresh = null;
+    private String runtimeErrMsg;
+
```

Comment:
You should add empty constructor 
```
public Dictionary() {}
```
The default value will be null if there is no empty constructor when deserializing.

Reply:
done